lib/puppet/provider/network_route/network_route.rb

```diff
@@ -0,0 +1,49 @@
+require 'net/ip'
+require 'ipaddr'
+#require_relative '../../../puppet_x/voxpupuli/utils'
+require 'puppet/resource_api/simple_provider'
+
+# Implementation for the network_route type using the Resource API.
+class Puppet::Provider::NetworkRoute::NetworkRoute < Puppet::ResourceApi::SimpleProvider
+  #include PuppetX::FileMapper
+
+  def routes_list
+    routes = []
+    Net::IP.routes.each do |route|
+      routes.push(route.instance_variables.each_with_object({}) { |var, hash| hash[var.to_s.delete("@")] = route.instance_variable_get(var) })
```

Comment:
Will definitely try and do that once I have the rest of the resource working.

So far it can get the routes on a system using the `net-ip` ruby gem (which in turn uses `ip route`)